src/java/org/apache/cassandra/metrics/RepairMetrics.java

```diff
@@ -18,17 +18,41 @@
 
 package org.apache.cassandra.metrics;
 
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+
 import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import org.apache.cassandra.net.Verb;
 
 import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
 
 public class RepairMetrics
 {
     public static final String TYPE_NAME = "Repair";
     public static final Counter previewFailures = Metrics.counter(DefaultNameFactory.createMetricName(TYPE_NAME, "PreviewFailures", null));
+    private static final Histogram retries = Metrics.histogram(DefaultNameFactory.createMetricName(TYPE_NAME, "Retries", null), false);
+    private static final Map<Verb, Histogram> retriesByVerb = Collections.synchronizedMap(new EnumMap<>(Verb.class));
```

Comment:
We have the thread-safety guarantees because `Verb` messages are never processed in parallel, right? `computeIfAbsent` doesn't look like a safe option for `synchronizedMap` because it's still not an atomic operation - so if messages for the `Verb` are not strictly ordered somewhere else we could lose metric values.

Reply:
yeah, I think `ConcurrentHashMap` is a better choice

Reply:
> We have the thread-safety guarantees because Verb messages are never processed in parallel, right?

we can process multiple repairs in parallel, so that isn't true.

> computeIfAbsent doesn't look like a safe option for synchronizedMap because it's still not an atomic operation 

`java.util.Collections.SynchronizedMap#computeIfAbsent` is the following on JDK 11

```
public V computeIfAbsent(K key,
                Function<? super K, ? extends V> mappingFunction) {
            synchronized (mutex) {return m.computeIfAbsent(key, mappingFunction);}
        }
```

@Mmuzaf where did you see it not use the lock?  This is thread safe on JDK 11...

> yeah, I think ConcurrentHashMap is a better choice

This is thread safe on JDK 11 and should have less memory, so the fact that CHM is faster with contention isn't as valuable to me as having less memory.

Reply:
@dcapwell My bad. I should have checked the `SynchronizedMap` implementation, not the `Map` interface itself, so you are right. How much RAM are we trying to save? I'm still concerned that we add the synchronized map to the hotpath, as my understanding is that metrics calculation should have no impact on the message processing as it is less important to the process itself.

Reply:
> I'm still concerned that we add the synchronized map to the hotpath

this isn't the hot path, this is retry messaging on the network thread... this is def the slow path =D

Also, your other feedback to preallocate avoids this problem as its now an immutable map...